distributed/worker.py

```diff
@@ -2081,7 +2081,10 @@ async def gather_dep(
                 stimulus_id=f"gather-dep-success-{time()}",
             )
 
-        except OSError:
+        # Note: CancelledError and asyncio.TimeoutError are rare conditions
+        # that can be raised by the network stack.
+        # See https://github.com/dask/distributed/issues/8006
+        except (OSError, asyncio.CancelledError, asyncio.TimeoutError):
```

Comment:
The catch-all below, which normally catches deserialization errors, is broken to begin with and will lead to a deadlocked cluster too. Fixing it is complicated because it would require implementing a memory->error transition on the scheduler.
- https://github.com/dask/distributed/issues/6705

Reply:
more in general I feel that we should treat everything that is a networking failure, which can and will happen in a production cluster, apart from (de)serialization error, internal errors, etc., which in theory should be deterministically weeded out in development.

Reply:
Thanks for clarifying!

Reply:
The catchall does not cover `CancelledError`s since those are `BaseException`s (since py3.8) https://docs.python.org/3/library/asyncio-exceptions.html#asyncio.CancelledError

This change is motivated because too many people were accidentally catching this exception which then caused hard to debug problems, see https://bugs.python.org/issue32528 for a conversation

I think that capturing `CancelledError` here is the correct way forward because `gather_dep` is not just a coroutine but we are always scheduling it as a dedicated `asyncio.Task`. This task terminates immediately in this exception handler and we're technically not even awaiting this task anywhere (this is the backend task foo in `BaseWorker`) so locking up due to not reraising is impossible.